app/forms/openid_connect_authorize_form.rb

```diff
@@ -62,7 +62,7 @@ def loa3_requested?
 
   def allowed_form_action
     return unless sp_redirect_uri =~ %r{https?://}
-    SecureHeadersWhitelister.extract_domain(sp_redirect_uri)
+    SecureHeadersWhitelister.extract_scheme_and_domain(sp_redirect_uri)
```

Comment:
Would it be easier to simply return `sp_redirect_uri` here, since it will automatically include the protocol? Is there a reason we don't want to include the full `redirect_uri` in the `form_action` directive?